functions/gravityForms/getGfFieldValidationSchema.js

```diff
@@ -1,41 +1,41 @@
 import getGfFieldId from '@/functions/gravityForms/getGfFieldId'
-import * as Yup from 'yup'
-
-// TODO Refactor this file for scaling purposes.
+import StringSchemaFactory from '@/functions/gravityForms/yupSchema/StringSchemaFactory'
 
 /**
- * Map props to validation schemas.
+ * Match field type with Yup schema object.
  *
- * @param {Object} fieldData GravityForm field props.
+ * Match GravityForm field to Yup API property.
+ *
+ * @author WebDevStudios
+ * @see https://github.com/jquense/yup#api
+ * @param {Object}   fieldData GravityForm field props.
+ * @returns {Object} Schema validation for field.
  */
-export default function getGfFieldValidationSchema(fieldData) {
-  const validationTypes = {
-    text: 'string'
-  }
+function getValidationSchemaByType(fieldData) {
+  let schemaGetter = null
 
-  const type = validationTypes?.[fieldData?.type]
+  switch (fieldData?.type) {
+    case 'text':
+      schemaGetter = new StringSchemaFactory(fieldData).schema
+      break
 
-  if (!type) {
-    return
+    default:
+      return
   }
 
-  const yupType = Yup[type]()
-
-  /**
-   * Conditionally concat Yup validations.
-   */
-  const validationSchema = yupType
-    .concat(
-      fieldData?.maxLength
-        ? yupType.max(
-            fieldData.maxLength,
-            `Must be ${fieldData.maxLength} characters or less`
-          )
-        : null
-    )
-    .concat(fieldData?.isRequired ? yupType.required('Required!') : null)
+  return schemaGetter
+}
 
+/**
+ * Map props to validation schemas.
+ *
+ * @param {Object}   fieldData GravityForm field props.
+ * @returns {Object} Schema validation for field.
+ *
+ * @author  WebDevStudios
```

Comment:
`@author` should be at the top.

Reply:
Good catch, I thought I had got all of those updated.